### ISSUE TITLE
Better instrument adding

### ISSUE THREAD

**Issue by ThatOnePianist** [musescore/MuseScore#20959]
### Your idea

So, when you start working on a new piece, whether it be a solo piano piece, a trio, or an orchestral piece, you are presented with this:
![image](https://github.com/musescore/MuseScore/assets/151048770/0d2036cc-6a28-4f19-984c-21d9223650ec)
Seems easy enough, right?
You click on the instrument you want, press on the arrow, and MuseScore automatically adds it to the score!
Ok, I want to have a piano... easy enough.
![image](https://github.com/musescore/MuseScore/assets/151048770/4fbf5320-0c5f-4c2b-a8b8-882d065a1e68)
Then, I want to have a trumpet in Bb...
![image](https://github.com/musescore/MuseScore/assets/151048770/a1c3e671-580c-4630-a766-6d37629eb021)
...and a clarinet in A.
Alright, a clarinet is a woodwind instrument, so...
![image](https://github.com/musescore/MuseScore/assets/151048770/9fbfaab6-f79f-4280-a7d0-d8b4798d178c)
Oh. MuseScore only has a clarinet in Bb. And that's what you might think.
...
And you'd be wrong.
Actually, MuseScore has clarinets in Bb, Eb, D, C, A and G.
But where are they?!
Sure, you can search for "clarinet" and..
![image](https://github.com/musescore/MuseScore/assets/151048770/7eb0f4f8-7ea9-4419-adda-92c74597f866)
There it is! Just click on the dropdown and select "A".
...
But why do you have to do it like this?
How about this:
If I want to add an instrument in another key (like a clarinet in A or C), I expect MuseScore to do this,:
![image](https://github.com/musescore/MuseScore/assets/151048770/45c4df0f-ed29-4b95-ae47-93112e8ee48f)
and not this:
![image](https://github.com/musescore/MuseScore/assets/151048770/4ad424df-18cf-43da-809e-78d9aaad61f2)
where, again, if I want a clarinet in A, I have to either search for "clarinet", or, do it the harder way: go to "all instruments", "woodwinds", scroll down a **ton** and find "clarinet".
It's ridiculous.

### Problem to be solved

Many orchestral writers' time.

**Comment by cbjeukendrup** [comment]
The confusing thing here, is that "Clarinet in A" is not in the "Common" instrument group (see the dropdown in the top left corner).
If you switch to the "All instruments" group, then "Clarinet in A" will show up in the same way as it does when you are searching.
Note that when you are searching, the dropdown automatically switches to "All instruments" temporarily.

Anyway, that's the logical explanation of the current behaviour, but that's not to say that the current behaviour is not confusing.

**Comment by MarcSabatella** [comment]
Even if these dropdowns were added, that only helps in these specific cases - adding the less common variants of the common instruments.  But you'd still need to use search or "All instruments" to find all the other instruments.  So no getting around the need for those, really.

But that said, I definitely like the idea of including these dropdowns where appropriate within the existing "Common" listing!

**Comment by shoogle** [comment]
I agree. If at least one of these related instruments is already visible then we might as well show the dropdown with the other options too.

Hopefully there is a straightforward way to do this programmatically in the code. If not, I can adjust the [instruments spreadsheet](https://docs.google.com/spreadsheets/d/1SwqZb8lq5rfv5regPSA10drWjUAoi65EuMoYtG-4k5s) so that the 'genres' are the same for all instruments with the same 'trackName'.

**Comment by bkunda** [comment]
I agree there's definitely scope for improvement here (so much so that I'd regard it as an improvement rather than a feature request). 
Thanks @ThatOnePianist.
